app/controllers/api/v1/lookup_keys_controller.rb

```diff
@@ -58,9 +58,10 @@ def update
 
       def destroy
         if @lookup_key.type == "PuppetclassLookupKey"
-          render_error 'unprocessable_entity', :status => :unprocessable_entity
+          @lookup_key.errors.add(:base, 'Smart class parameters cannot be destroyed')
```

Comment:
I don't think this should be in a controller...
override the destroy method on PuppetclassLookupKey. Alias it to destroy! (if no such method exists already), so we can destroy if we really need to.